cocaine/proxy/proxy.py

```diff
@@ -692,6 +738,11 @@ def stop_condition(body):
                 else:
                     raise Exception("unsupported X-Cocaine-HTTP-Proto-Version: %s" % cocaine_http_proto_version)
 
+                if headers.get('Content-Length'):
+                    processor = CachedBodyProcessor(request, name, code, headers)
+                else:
+                    processor = ChunkedBodyProcessor(request, name, code, headers)
```

Comment:
Seems like code duplication here. You need only to determine a class type to instantiate.